java/server/src/org/openqa/selenium/grid/sessionmap/jdbc/JdbcBackedSessionMap.java

```diff
@@ -0,0 +1,183 @@
+
+
+  public JdbcBackedSessionMap(Tracer tracer, Connection jdbcConnection, EventBus bus)  {
+    super(tracer);
+
+    Require.nonNull("JDBC Connection Object", jdbcConnection);
+    this.bus = Require.nonNull("Event bus", bus);
+
+    this.connection = jdbcConnection;
+    this.bus.addListener(SESSION_CLOSED, event -> {
+      SessionId id = event.getData(SessionId.class);
+      remove(id);
+    });
+  }
+
+  public static SessionMap create(Config config) {
+    Tracer tracer = new LoggingOptions(config).getTracer();
+    EventBus bus = new EventBusOptions(config).getEventBus();
+
+    JdbcSessionMapOptions sessionMapOptions = new JdbcSessionMapOptions(config);
+
+    Connection connection;
+
+    try {
+      connection = sessionMapOptions.getJdbcConnection();
+    } catch (SQLException e) {
+      throw new ConfigException(e.toString());
+    }
+
+    return new JdbcBackedSessionMap(tracer, connection, bus);
+  }
+  @Override
+  public boolean add(Session session) {
+    Require.nonNull("Session to add", session);
+
+    try {
+      return insertSessionStatement(session).executeUpdate() >= 1;
+
+    } catch (SQLException e) {
+      throw new JdbcException(e.getMessage());
+    }
+  }
+
+  @Override
+  public Session get(SessionId id) throws NoSuchSessionException {
+    Require.nonNull("Session ID", id);
+
+    URI uri = null;
+    Capabilities caps = null;
+    String rawUri = null;
+
+    try (ResultSet sessions = readSessionStatement(id).executeQuery()){
+      if (sessions.next()) {
+        rawUri = sessions.getString(SESSION_URI_COL);
+        String rawCapabilities = sessions.getString(SESSION_CAPS_COL);
+
+        caps = rawCapabilities == null ?
+                            new ImmutableCapabilities() :
+                            JSON.toType(rawCapabilities, Capabilities.class);
+        try {
+          uri = new URI(rawUri);
+        } catch (URISyntaxException e) {
+          throw new NoSuchSessionException(String.format("Unable to convert session id (%s) to uri: %s", id, rawUri), e);
+        }
+      } else {
+        throw new NoSuchSessionException("Unable to find URI for session " + id);
+      }
+
+      return new Session(id, uri, caps);
+    } catch (SQLException e) {
+      throw new JdbcException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void remove(SessionId id) {
+    Require.nonNull("Session ID", id);
+
+    try {
+      getDeleteSqlForSession(id).executeUpdate();
+    } catch (SQLException e) {
+      throw new JdbcException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void close() {
+    try {
+      connection.close();
+    } catch (SQLException e) {
+      LOG.warning("SQL exception while closing JDBC Connection:" + e.getMessage());
+    }
+  }
+
+  private PreparedStatement insertSessionStatement(Session session) throws SQLException {
+    PreparedStatement insertStatement = connection.prepareStatement("insert into " + TABLE_NAME
```

Comment:
Nit: Newline before the arguments begin. That avoids excessive line length and makes the code a bit easier to read.